chartfx-dataset/src/main/java/de/gsi/dataset/serializer/IoClassSerialiser.java

```diff
@@ -0,0 +1,458 @@
+                    return (FieldSerialiser<E>) entry;
+                }
+            }
+            // found FieldSerialiser entry but not matching required generic types
+        }
+
+        // did not find FieldSerialiser entry by specific class -> search for assignable interface definitions
+
+        final List<Class<?>> potentialMatchingKeys = new ArrayList<>(10);
+        for (Type key : knownClasses().keySet()) {
+            final Class<?> testClass = ClassUtils.getRawType(key);
+            if (testClass.isAssignableFrom(clazz)) {
+                potentialMatchingKeys.add(testClass);
+            }
+        }
+        if (potentialMatchingKeys.isEmpty()) {
+            // did not find any matching clazz/interface FieldSerialiser entries
+            return null;
+        }
+
+        final List<FieldSerialiser<?>> interfaceMatchList = new ArrayList<>(10);
+        for (Class<?> testClass : potentialMatchingKeys) {
+            final List<FieldSerialiser<?>> fieldSerialisers = knownClasses().get(testClass);
+            if (fieldSerialisers.isEmpty()) {
+                continue;
+            }
+            interfaceMatchList.addAll(fieldSerialisers);
+        }
+        if (interfaceMatchList.size() == 1 || classGenericArguments == null || classGenericArguments.isEmpty()) {
+            // found single match FieldSerialiser entry type w/o specific generics requirements
+            return (FieldSerialiser<E>) interfaceMatchList.get(0);
+        }
+
+        // more than one possible serialiser implementation
+        for (final FieldSerialiser<?> entry : interfaceMatchList) {
+            if (checkClassCompatibility(classGenericArguments, entry.getGenericsPrototypes())) {
+                // found generics matching or assignable entry
+                return (FieldSerialiser<E>) entry;
+            }
+        }
+        // could not match with generics arguments
+
+        // find generic serialiser entry w/o generics parameter requirements
+        return (FieldSerialiser<E>) interfaceMatchList.stream().filter(entry -> entry.getGenericsPrototypes().isEmpty()).findFirst().orElse(null);
+    }
+
+    public IoSerialiser getIoSerialiser() {
+        return ioSerialiser;
+    }
+
+    public final <E> BiFunction<Type, Type[], FieldSerialiser<E>> getSerialiserLookupFunction() {
+        return (primaryType, secondaryType) -> {
+            if (primaryType == null) {
+                throw new IllegalArgumentException("no serialiser implementation found for classType = " + primaryType);
+            }
+            return (FieldSerialiser<E>) cacheFindFieldSerialiser(ClassUtils.getRawType(primaryType), secondaryType == null ? Collections.emptyList() : Arrays.asList(secondaryType));
+        };
+    }
+
+    public Map<Type, List<FieldSerialiser<?>>> knownClasses() {
+        return classMap;
+    }
+
+    public void serialiseObject(final Object rootObj, final ClassFieldDescription classField, final int recursionDepth) {
```

Comment:
Codacy found an issue: [The method 'serialiseObject(Object, ClassFieldDescription, int)' has an NPath complexity of 1080, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)